cherrypy/wsgiserver/__init__.py

```diff
@@ -1695,7 +1695,7 @@ def stop(self, timeout=5):
 
         # Don't join currentThread (when stop is called inside a request).
         current = threading.currentThread()
-        if timeout and timeout >= 0:
+        if timeout is not None and timeout >= 0:
```

Comment:
Could you please extend existing tests with corresponding test cases for negative/positive testing?